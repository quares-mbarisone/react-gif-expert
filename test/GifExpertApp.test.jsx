import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp"

describe('Pruebas en <GifExpertApp/>', () => {
    
    const inputValue = 'One Piece'
    
    test('Debe de poder describirse en el input', () => {
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox');

        fireEvent.input( input, { target: { value: inputValue } } );

        expect( input.value ).toBe(inputValue)
    })

    test('Debe de poder mandarse el formulario', () => {
        render( <GifExpertApp /> )
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, { target: { value: inputValue } } );
        
        expect( input.value ).toBe(inputValue);
        
        fireEvent.submit( form );

        expect( input.value ).toBe('');
    })

    test('debe de hacer match con la snapshot', () => {
        const { container } = render( <GifExpertApp /> )
        expect( container ).toMatchSnapshot();
    })

    test('No debe agregar la categoría si ya existe', () => {
        render( <GifExpertApp /> )
        
    })

    test('Debe agregar la categoría si no existe', () => {
        render( <GifExpertApp /> )
    })

})