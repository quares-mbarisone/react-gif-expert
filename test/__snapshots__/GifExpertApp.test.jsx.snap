// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe de hacer match con la snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Busca tu Gif"
      type="text"
      value=""
    />
  </form>
  <h3>
     
    One Piece
     
  </h3>
  <h2>
     Cargando... 
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
